lib/Core/Executor.cpp

```diff
@@ -1935,27 +1935,43 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
     // Arithmetic / logical
 
   case Instruction::Add: {
+    assert(i->getOperand(0)->getType()->isIntegerTy() &&
```

Comment:
I agree with @ccadar and @delcypher ;)
Too many checks, it clutters the code a lot. Can this be improved?